addons/account_analytic_default/views/account_analytic_default_view.xml

```diff
@@ -74,29 +73,29 @@
             <field name="res_model">account.analytic.default</field>
             <field name="context">{'search_default_product_id': [active_id], 'default_product_id': active_id}</field>
         </record>
-        <record model="ir.ui.view" id="product_form_view_default_analytic_button">
+        <record id="product_form_view_default_analytic_button" model="ir.ui.view">
```

Comment:
split this product and product.template changes in separate view file same like which you did for python model